pkg/cmd/release/download/download.go

```diff
@@ -108,11 +139,25 @@ func downloadRun(opts *DownloadOptions) error {
 	}
 
 	var toDownload []shared.ReleaseAsset
-	for _, a := range release.Assets {
-		if len(opts.FilePatterns) > 0 && !matchAny(opts.FilePatterns, a.Name) {
-			continue
+	// create Asset with ZipBallURL or TarBallURL if using archive option
+	if len(opts.ArchiveType) > 0 {
+		var defaultUrl = release.ZipballURL
```

Comment:
Let's call this archiveURL:
```suggestion
		archiveURL := release.ZipballURL
```